src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,25 +42,34 @@ export const TodoModal: React.FC = () => {
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              Todo #{id}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => onClose(undefined)}
```

Comment:
The naming convention for event handler functions should follow the pattern `handleEventName`. Consider renaming the inline function to `handleClose` to align with the recommended naming conventions.